src/ngraph_builder.cc

```diff
@@ -1716,20 +1727,38 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
       }
       auto lower_vec = ng_begin_const->get_vector<int>();
 
-      auto ng_size_const =
-          std::dynamic_pointer_cast<ng::op::Constant>(ng_size->second);
-      if (ng_size_const == nullptr) {
+      auto ng_end_const =
+          std::dynamic_pointer_cast<ng::op::Constant>(ng_end->second);
+      if (ng_end_const == nullptr) {
         return tf::errors::InvalidArgument(
             "The argument size is null for StridedSlice");
       }
-      auto size_vec = ng_size_const->get_vector<int>();
+      auto end_vec = ng_end_const->get_vector<int>();
 
-      auto& input_shape = ng_input->second->get_shape();
       NGRAPH_VLOG(3) << "Begin input for StridedSlice: " << ng::join(lower_vec);
-      NGRAPH_VLOG(3) << "Size input for StridedSlice: " << ng::join(size_vec);
-      if (std::any_of(size_vec.begin(), size_vec.end(), [](int i){ return i <= 0; })) {
-        std::transform(size_vec.begin(), size_vec.end(), input_shape.begin(),
-                       size_vec.begin(), [](int first, int second) {
+      NGRAPH_VLOG(3) << "End input for StridedSlice: " << ng::join(end_vec);
+
+      auto& input_shape = ng_input->second->get_shape();
+      NGRAPH_VLOG(3) << "Input shape for StridedSlice: " << ng::join(input_shape);
+
+      if (tf_shrink_axis_mask != 0) {
```

Comment:
From what I can tell from the docs, `shrink_axis_mask` is a bit vector where a "1" indicates that an axis, having been chopped down to one dimension, should be removed from the output shape.

Reply:
Can we keep the stuff like -1 means from the end kind of stuff out of our ops and let the python bindings deal with all the creeping featurism in the python versions?

Reply:
@diyessi Sorry, missed this. For sure---this is the point where we translate's TensorFlow's fancy ops to nGraph's RISC-y ones.